test/test_rpc.py

```diff
@@ -0,0 +1,93 @@
+import sys
+import tempfile
+import time
+
+from functools import wraps
+
+
+import torch
+import torch.distributed as dist
+import torch.multiprocessing as mp
+
+from common_distributed import MultiProcessTestCase
+from common_utils import TestCase, load_tests, run_tests
+from common_utils import NO_MULTIPROCESSING_SPAWN
+
+
+# load_tests from common_utils is used to automatically filter tests for
+# sharding on sandcastle. This line silences flake warnings
+load_tests = load_tests
+
+if not dist.is_available():
+    print('c10d not available, skipping tests')
+    sys.exit(0)
+
+
+def _init_rpc(func):
+    def wrapper(self):
+        store = dist.FileStore(self.file.name, self.world_size)
+        dist.init_process_group(backend='gloo', rank=self.rank,
+                                world_size=self.world_size, store=store)
+        dist.init_rpc('worker%d' % self.rank)
+        func(self)
+        dist.destroy_rpc()
+        dist.destroy_process_group(dist.group.WORLD)
+
+    return wrapper
+
+class RpcTest(MultiProcessTestCase):
+
+    @property
+    def world_size(self):
+        return 4
+
+    @_init_rpc
+    def test_add(self):
+        n = self.rank + 1
+        dstRank = n % self.world_size
+        ret = dist.rpc_sync('worker%d' % dstRank, torch.add,
+                            args=(torch.ones(n, n), torch.ones(n, n)))
+        dist.barrier()
```

Comment:
In the commit on Aug 2, 2019,

I noticed that 3 methods, `test_scalar_add `, `test_async_add ` and `test_nonzero` have exceptions on exiting, because they didn't call `dist.barrier()`.

While `test_add` and `test_multi_rpc` always passed because they called ` dist.barrier()`.

Should we remove `dist.barrier()` in test methods?

and leave this to `dist.destroy_rpc()`, which is called in `_init_rpc ` decorator.

Reply:
Yes, one minor thing is that. It can either be put into `shutdown` or `RpcAgent` needs to expose which backend it is using, as `dist.barrier()` only works for `pg` backend.